public/app/percona/dbaas/components/Kubernetes/Kubernetes.hooks.ts

```diff
@@ -81,10 +92,37 @@ export const useKubernetes = (): [Kubernetes[], DeleteKubernetesAction, AddKuber
   return [kubernetes, deleteKubernetes, addKubernetes, loading];
 };
 
-const toModelList = (response: KubernetesListAPI): Kubernetes[] => (response.kubernetes_clusters ?? []).map(toModel);
+const toModelList = (response: KubernetesListAPI, checkUpdateResponse: CheckOperatorUpdateAPI): Kubernetes[] =>
+  (response.kubernetes_clusters ?? []).map(toModel(checkUpdateResponse));
 
-const toModel = (response: KubernetesAPI): Kubernetes => ({
-  kubernetesClusterName: response.kubernetes_cluster_name,
-  operators: response.operators,
-  status: response.status as KubernetesClusterStatus,
+const toModel = (checkUpdateResponse: CheckOperatorUpdateAPI) => ({
+  kubernetes_cluster_name: kubernetesClusterName,
+  operators,
+  status,
+}: KubernetesAPI): Kubernetes => ({
+  kubernetesClusterName,
+  operators: toModelOperators(kubernetesClusterName, operators, checkUpdateResponse),
+  status: status as KubernetesClusterStatus,
 });
+
+// adds avaiableVersion to operators dynamically
+const toModelOperators = (
+  kubernetesClusterName: string,
+  operators: OperatorsList,
+  { cluster_to_components }: CheckOperatorUpdateAPI
+): OperatorsList => {
+  const modelOperators = {} as OperatorsList;
+  const componentToUpdate = cluster_to_components[kubernetesClusterName].component_to_update_information;
+
+  Object.entries(operators).map(([operatorKey, operator]: [keyof OperatorsList, Operator]) => {
```

Comment:
maybe this should be `forEach` instead of `map`, as we aren't saving any transformed array

Reply:
Totally, forgot to change it! Done!